File: baekjoon_all/09000+/boj_9012.py
# Solve 2022-05-28
# Update 2025-09-05

import sys

input = lambda: sys.stdin.readline().rstrip()

def validate_parentheses(ps: str) -> bool:
    depth = 0

    for par in ps:
        if par == "(":
            depth += 1
        else:
            depth -= 1

            if depth < 0:
                return False

    return depth == 0


for _ in range(int(input())):
    ps = input()
    print("YES" if validate_parentheses(ps) else "NO")

File: baekjoon_all/09000+/boj_9012.java
// Solve 2024-01-21
// Update 2025-07-27

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class boj_9012 {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        StringBuilder sb = new StringBuilder();
        int t = Integer.parseInt(br.readLine());

        while (t-- > 0) {
            String parentheses = br.readLine();
            sb.append(isParenthesesValid(parentheses) ? "YES\n" : "NO\n");
        }

        System.out.print(sb);
    }

    static boolean isParenthesesValid(String parentheses) {
        int depth = 0;

        for (int i = 0; i < parentheses.length(); i++) {
            char par = parentheses.charAt(i);

            if (par == '(') {
                depth++;
            } else {
                depth--;

                if (depth < 0) return false;
            }
        }

        return depth == 0;
    }

}
